docker-compose.yml

```
@@ -0,0 +1,79 @@
+services:
+  db:
+    image: postgis/postgis:17-3.5-alpine
+    restart: always
+    environment:
+      POSTGRES_PASSWORD: S5mhPe(+6CvWZbhj
+      POSTGRES_DB: anacarde
+      POSTGRES_USER: anacarde
+    ports:
+      - "5432:5432"
+    volumes:
+      - anacarde-database:/var/lib/postgresql/data
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U anacarde -d anacarde"]
+      interval: 30s
+      timeout: 60s
+      retries: 5
+      start_period: 80s
+
+  pgadmin:
+    image: dpage/pgadmin4:latest
+    restart: always
+    environment:
+      PGADMIN_DEFAULT_EMAIL: admin@example.com
+      PGADMIN_DEFAULT_PASSWORD: adminpassword
+    ports: 
+      - "8081:80"
+    depends_on:
+      db: 
+        condition: service_healthy
+    volumes: 
+      - ./api/servers.json:/pgadmin4/servers.json
+
+  mailpit:
+    image: axllent/mailpit:latest
+    restart: unless-stopped
+    ports: 
+      - "1025:1025" 
+      - "8025:8025" 
+    volumes: 
+      - mailpit-data:/data
+    environment:
+      MP_MAX_MESSAGES: 5000
+      MP_DATABASE: /data/mailpit.db
+      MP_SMTP_AUTH_ACCEPT_ANY: 1
+      MP_SMTP_AUTH_ALLOW_INSECURE: 1
+    healthcheck:
+      test: ["CMD-SHELL", "curl -f http://localhost:8025 || exit 1"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+
+  api:
+    build:
+      context: ./api
+      dockerfile: Dockerfile
+    depends_on:
+      db: { condition: service_healthy }
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:postgresql://db:5432/anacarde
+      SPRING_DATASOURCE_USERNAME: anacarde
+      SPRING_DATASOURCE_PASSWORD: S5mhPe(+6CvWZbhj
+      SPRING_MAIL_HOST: mailpit
+      SPRING_MAIL_PORT: 1025
+    expose:
+      - "8080" 
```

ça ne fonctionnait pas car on ne pouvait pas joindre l'api en fait!
```suggestion
    ports:
      - "8080:8080" 
```

---

J'ajoute ça aussi qui je pense devrait être ajouté, ça permettrait de diminuer le traffic.

```
https://bootcamptoprod.com/brotli-compression-in-spring-boot/
```
en GZIP c'est plus simple à mettre en place mais brotli est meilleurs sur tous les points: 

```
server:
  compression:
    enabled: true
    mime-types: text/html,application/json,application/xml,text/css,text/javascript,application/javascript
    min-response-size: 1024  # ne compresser que si > 1 Ko
```

Note le minimum response size qui est intéressant, je ne pense pas que le type check ça dans son filtre.

J'aimerais bien une petite section dans le rapport sur un peu tout ça. Je trouve perso cela intéressant, l'idée est de réduire l'utilisation au maximum de la bande passante, critique au Bénin.

Tu peux aussi parler de zStandard: 

- https://paulcalvano.com/2024-03-19-choosing-between-gzip-brotli-and-zstandard-compression/

Et pourquoi on ne l'a pas utilisé: 

- https://caniuse.com/brotli
- https://caniuse.com/?search=z%20standard